lacework/resource_lacework_integration_docker_hub.go

```diff
@@ -145,30 +114,15 @@ func resourceLaceworkIntegrationDockerHub() *schema.Resource {
 
 func resourceLaceworkIntegrationDockerHubCreate(d *schema.ResourceData, meta interface{}) error {
 	lacework := meta.(*api.Client)
-
-	limitByTags := d.Get("limit_by_tag").(string)
-	if tags := castAttributeToStringSlice(d, "limit_by_tags"); len(tags) != 0 {
-		limitByTags = strings.Join(tags, ",")
-	}
-
-	limitByLabels := d.Get("limit_by_label").(string)
-	if labels := castAttributeToStringKeyMapOfStrings(d, "limit_by_labels"); len(labels) != 0 {
-		limitByLabels = joinMapStrings(labels, ",")
-	}
-
-	limitByRepos := d.Get("limit_by_repos").(string)
-	if repos := castAttributeToStringSlice(d, "limit_by_repositories"); len(repos) != 0 {
-		limitByRepos = strings.Join(repos, ",")
-	}
-
-	data := api.NewDockerHubRegistryIntegration(d.Get("name").(string),
-		api.ContainerRegData{
-			LimitByTag:       limitByTags,
-			LimitByLabel:     limitByLabels,
-			LimitByRep:       limitByRepos,
+	data := api.NewContainerRegistry(d.Get("name").(string),
+		api.DockerhubContainerRegistry,
+		api.DockerhubData{
+			LimitByTag:       castAttributeToStringSlice(d, "limit_by_tags"),
+			LimitByRep:       castAttributeToStringSlice(d, "limit_by_repositories"),
+			LimitByLabel:     castAttributeToArrayKeyMapOfStrings(d, "limit_by_labels"),
```

Comment:
Why did we use this function? 🤔 This makes no sense to me.

Reply:
The Schema for this field is:
```
                "limitByLabel": {
                  "default": [],
                  "description": "When sending a request, if you do not want to assess all images in this registry, specify `key:value` pairs so that only images with matching label `key:value` pairs will be assessed. To change which images you want to assess, update this field so the change is captured during the next polling period. Supported field input: `key:value`. If you specify tag and label limits, they function as an AND. When included in a response, returns a list of labels that will be used to assess images.",
                  "items": {
                    "example": [
                      {
                        "key1": "value1"
                      },
                      {
                        "key2": "value2"
                      }
                    ],
                    "type": "object"
                  },
                  "title": "Limit by Label(s) (optional)",
                  "type": "array"
                },
```